tfjs-core/src/ops/einsum_test.ts

```diff
@@ -262,4 +262,26 @@ describeWithFlags('einsum', ALL_ENVS, () => {
     const y = tensor2d([[0, 1], [2, 3], [4, 5]]);
     expect(() => tf.einsum('ij,jk->ik->i', x, y)).toThrowError(/exactly one/);
   });
+
+  it('reduce einsum to batch matmul', async () => {
```

Comment:
how does this test ensure it is reduce to matmul op? it only check the results are the same, right?

Reply:
Decoupled `checkReducibleToMatMul` out of einsum implementation to ensure this.

An alternative approach is to make `spyOn` to tf.matMul  to check if matMul is called. However, the methods/properties of module `tf` is immutable, so we could not use `spyOn(tf, 'matMul')`. To make this work, we need to have some mechanism similar as tfjs-converter's op-executor, which allows to insert [fake dependency](https://github.com/tensorflow/tfjs/blob/master/tfjs-converter/src/operations/executors/arithmetic_executor_test.ts#L74). However, the efforts are non-trivial.